app/tests/e2e/test_wallet_credentials.py

```diff
@@ -60,3 +61,32 @@ async def test_get_and_delete_credential_record(
             f"{WALLET_CREDENTIALS_PATH}/{credential_id}"
         )
     assert exc.value.status_code == 404
+
+
+@pytest.mark.anyio
+async def test_get_credential_record_with_limit(
+    alice_member_client: RichAsyncClient,
+    issue_alice_creds_non_revoke: List[  # pylint: disable=unused-argument
+        CredentialExchange
+    ],
+):
+
+    credentials_1 = (
+        await alice_member_client.get(f"{WALLET_CREDENTIALS_PATH}?limit=5")
+    ).json()
+
+    assert len(credentials_1["results"]) == 5
+
+    credentials_2 = (
+        await alice_member_client.get(f"{WALLET_CREDENTIALS_PATH}?limit=5&offset=5")
+    ).json()
+
+    assert len(credentials_2["results"]) == 5
+    assert credentials_1 != credentials_2
+
+    credentials_3 = (
+        await alice_member_client.get(f"{WALLET_CREDENTIALS_PATH}?limit=5")
+    ).json()
+
+    assert len(credentials_3["results"]) == 5
+    assert credentials_1 == credentials_3
```

Comment:
There's a few more cases to cover.

Let's say it's changed to just 3 credentials. We want to assert: limit 1,2,3 all return expected, and limit 4 returns 3.
Then, limit 1 + offset 3 should return an empty list.

For completeness we can also assert invalid params:
```py
        # Test invalid limit and offset values
        invalid_params = [
            {"limit": -1},  # must be positive
            {"offset": -1},  # must be positive
            {"limit": 0},  # must be greater than 0
            {"limit": 10001},  # must be less than or equal to max in ACA-Py: 10'000
        ]

        for params in invalid_params:
            with pytest.raises(HTTPException) as exc:
                await alice_member_client.get(BASE_PATH, params=params)
            assert exc.value.status_code == 422
```

Look to the `test_get_connections_paginated` for what assertions we currently do on paginated records